check-code-style/action.yml

```diff
@@ -6,21 +6,39 @@ desciption: 'Check files for correct code style'
 runs:
   using: "composite"
   steps:
-    - name: Run bootstrap
-      run: cd dev-tools/ && ./bootstrap
-      # The `shell` section is mandatory. Without it we will get
-      # the following error:
-      # ***  Required property is missing: shell.  
+    # IMPORTANT NOTE.
+    # https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md
+    # If you check the link above you will see that on Ubuntu 20.04
+    # we have preinstalled clang-format-10/11/12. By default the symlink
+    # to /usr/bin/clang-format-11 is set. That's why we will not be able
+    # to run successfully the checks for code style cause the minimum
+    # version required is `12` (see `dev-tools/check-code-style/check_style.sh` 
+    # especially check_clang_format() function).
+    # That's why the main idea of this step is just setting the symlink to
+    # /usr/bin/clang-format-12, nothing more. Thus we will be able to run all
+    # checks with correct version of clang-format using GitHub Actions. 
+    - name: Create symlink to the newest version of clang-format
+      run: |
+        # Before we recreate the symlink we should delete
+        # the existing one.
+        sudo rm /etc/alternatives/clang-format
+
+        # Recreate symlink 'clang-format'
+        sudo ln -s /usr/bin/clang-format-12 /etc/alternatives/clang-format 
```

Comment:
You've still got a global invariant here! The invariant is that the caller will only call this from a GitHub Workflow that uses ubuntu. Instead this action should just assume that it can call `clang-format`, or you should use an [input parameter](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs).